### DISCUSSION TITLE
Error upgrading to version greater than 21.02

### DISCUSSION THREAD

**Discussion by samsisamsi** [webmin/authentic-theme#1675] [General]
When I try to update Authentic Theme to version 21.02 or greater, I get the error message:
HTTP/1.0 500 Perl execution failed Server: MiniServ/2.100 Date: Wed, 20 Sep 2023 15:24:30 GMT Content-type: text/html; Charset=utf-8 Connection: close
Error — Perl execution failed
Undefined subroutine &virtual_server::get_module_version_and_type called at /usr/share/webmin/authentic-theme/authentic-lib.pl line 699.
Any ideas?
OS: Ubuntu 22.04.3

**Comment by iliaross** [comment]
Try installing the latest Webmin 2.102 first.

**Comment by samsisamsi** [comment]
When I tried to install Webmin 2.102 over Webmin 2.100, I get the same error. So I had to downgrade again to 2.100. It seems that upgrading Webmin from 2.100 to 2.102 automatically upgrades Authentic Theme to the latest version.

**Comment by iliaross** [comment]
Do you have Virtualmin installed, if so which version?

**Comment by samsisamsi** [comment]
I have Virtualmin installed, but never used and probably never will. Version is 6.05.

**Comment by iliaross** [comment]
Got it! I have just made a patch to Authentic Theme. You could go to Theme Configuration page and install latest development version to work around your problem.

**Comment by samsisamsi** [comment]
It worked. Thanks!
